test/simdat.jl

```diff
@@ -39,3 +39,160 @@ using Test
+
+    @test nrow(data) == subj_n * (item_n/cond_n)
+
+    # Test whether each subject is only in one of the conditions
+    for s in nlevels(subj_n,"S")
+        @test length(unique(data[isequal.(data.subj, s), :cond])) == 1
+    end
+
+    # Test whether each item is only in one of the conditions
+    for i in nlevels(item_n,"I")
+        @test length(unique(data[isequal.(data.item, i), :cond])) == 1
+    end
+
+end;
+
+@testset "simdat_crossed between-subjects between-items - complex case" begin
+    # stimulate data for a case in which a factor is both between-subject and between-item
+    subj_btwn = Dict(:age => ["O", "Y"],
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    item_btwn = Dict(:pet => ["cat", "dog"],
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    both_win = Dict(:time => ["morning", "evening"])
+
+    subj_n = 12
+    item_n = 12
+
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+        subj_btwn = subj_btwn,
+        item_btwn = item_btwn,
+        both_win = both_win))
+
+    # Test whether each subject is only in one of the levels of the between-subject/between-items conditions
+    for s in nlevels(subj_n, "S"), cond in [:cond1, :cond2]
+        @test length(unique(data[isequal.(data.subj, s), cond])) == 1
+    end
+
+    # Test whether each item is only in one of the conditions
+    for i in nlevels(item_n, "I"), cond in [:cond1, :cond2]
+        @test length(unique(data[isequal.(data.item, i), cond])) == 1
+    end
+
+end;
+
+@testset "simdat_crossed test all combinations" begin
+        subj_btwn = Dict(:age => ["O", "Y"], # -> item within
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    item_btwn = Dict(:pet => ["cat", "dog"], # -> subject within
+                     :cond1 => ["A", "B"],
+                     :cond2 => ["C", "D", "E"])
+    both_win = Dict(:time => ["morning", "evening"])
+
+
+    subj_n = 12
+    item_n = 12
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+        subj_btwn=subj_btwn,
+        item_btwn=item_btwn,
+        both_win=both_win))
+    s2 = subset(data, :subj => x -> x .== "S02")    
+    @test all(s2.cond1 .== "A")
+    @test all(s2.age .== "Y")
+    @test length(unique(s2.pet)) == 2 # from item between
+    
+    #-----
+    # no subject between
+    data = DataFrame(simdat_crossed(subj_n, item_n,
+        item_btwn=item_btwn,
+        both_win=both_win))
+
+    @test nrow(data) == 288 # many more rows because many more effects are within-subject
+    s2 = subset(data, :subj => x -> x .== "S02")
+    @test length(unique(string.(s2.cond1) .* string.(s2.cond2))) == 6 # test all 6 combinations are there
+    @test length(unique(s2.cond1)) == 2 # now we have both conditions within subject
+
+    i2 = subset(data, :item => x -> x .== "I02")
+    @test all(i2.cond1 .== "A") # but only one because cond1 is between-items here
```

Comment:
```suggestion
    i2 = subset(data, :item => ByRow(==("I02")))
    @test all(==("A"), i2.cond1) # but only one because cond1 is between-items here
```